Fix name collisions

mjd3 opened the pull request:
- Fixes #49 by checking AWS similar to `list` and regenerating until the collision is resolved
- Removed unused files, deps, and classes

Note: Currently, name collisions are only checked in the default region; we may want to extend to all regions?

jeffi commented on fogros2/fogros2/aws.py:
```diff
@@ -252,16 +235,31 @@ def __init__(self, ami_image, region="us-west-1", ec2_instance_type="t2.micro",
+                tag_map = {t["Key"]: t["Value"] for t in inst["Tags"]} if "Tags" in inst else {}
+                if "FogROS2-Name" in tag_map:
+                    cur_instance_names.append(tag_map["FogROS2-Name"])
+        while self.name in cur_instance_names:
```
This is a race-condition, but extremely unlikely to occur.  If another fogros concurrently starts with the same name between line 244 and here, it will result in an undetected name collision here.

mjd3 replied:
Agree; I felt the tradeoff for speed by not querying every time was worth it. Plus we don't actually get knowledge of the new instance until it starts, so the condition you're mentioning is still possible even by querying every time if someone starts two instances at roughly the same time (both would query and not find the collision).

mjd3 replied:
This is resolved in the updated version

jeffi commented on fogros2/fogros2/aws.py:
```diff
@@ -252,16 +235,31 @@ def __init__(self, ami_image, region="us-west-1", ec2_instance_type="t2.micro",
         self.ec2_boto3_client = boto3.client("ec2", self.region)
 
+        # Check for name collision among other AWS instances
+        cur_instances = self.ec2_boto3_client.describe_instances(
```
Did you explore seeing if you could get AWS to reject a duplicate name?


[Jeff Ichnowski](https://app.slack.com/team/UK66LLFHN)  [15 minutes ago](https://learningsystems.slack.com/archives/C01MZN8ETMH/p1652320759477599?thread_ts=1652320383.335449&cid=C01MZN8ETMH)
thus have it be 1 AWS call to create in the normal case


[Jeff Ichnowski](https://app.slack.com/team/UK66LLFHN)  [14 minutes ago](https://learningsystems.slack.com/archives/C01MZN8ETMH/p1652320768364969?thread_ts=1652320383.335449&cid=C01MZN8ETMH)
and loop on failure in the exception case?


Jeff Ichnowski  14 minutes ago
right now it looks like its doing 2 calls in the normal case--the first call gets a list of instances, which could be a long list (some people have 10000s of instances)

mjd3 replied:
According to what I could find, the "name" tag is not used by AWS at all (since it relies on the internal instance IDs), so I don't think there is a good way to autoreject the creation when it has the same name. Instead, I query just for instances with the same name (not all instances like before) and keep generating new names until we get one that isn't found in the system. Also added in the key pair deleting (this can be done without issue, see code). Let me know if you have any further thoughts!

jeffi reviewed (commented):
See inline comments below.

mjd3 commented:
Added key pair auto deleting if there's a collision on that front (since instance would already be terminated) and some other cleaning up (e.g., VPN was in two files, some variables were not used anywhere, converted some to properties if appropriate, made most internal variables to be prefaced with `_`)
